File: src/unitysvc_data/examples/llm/code-example-requests/code-example-v1.py
import os

import requests

UNITYSVC_API_KEY = os.environ["UNITYSVC_API_KEY"]
SERVICE_BASE_URL = os.environ["SERVICE_BASE_URL"]
MODEL = os.environ["MODEL"]

response = requests.post(
    SERVICE_BASE_URL,
    headers={
        "Authorization": f"Bearer {UNITYSVC_API_KEY}",
        "Content-Type": "application/json",
    },
    json={
        "model": MODEL,
        "messages": [
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "user", "content": "Say this is a test"},
        ],
    },
)
response.raise_for_status()
print(response.json()["choices"][0]["message"]["content"])

File: src/unitysvc_data/examples/llm/code-example-javascript/code-example-v1.js
const { UNITYSVC_API_KEY, SERVICE_BASE_URL, MODEL } = process.env;

async function main() {
  const response = await fetch(`${SERVICE_BASE_URL}/chat/completions`, {
    method: "POST",
    headers: {
      Authorization: `Bearer ${UNITYSVC_API_KEY}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      model: MODEL,
      messages: [
        { role: "system", content: "You are a helpful assistant." },
        { role: "user", content: "Say this is a test" },
      ],
    }),
  });
  if (!response.ok) {
    throw new Error(`HTTP ${response.status}: ${await response.text()}`);
  }
  const data = await response.json();
  console.log(data.choices[0].message.content);
}

main().catch((err) => {
  console.error(err.message);
  process.exit(1);
});
